akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSourceSpec.scala

```diff
@@ -290,6 +291,15 @@ class QueueSourceSpec extends StreamSpec {
           .expectComplete()
         source.watchCompletion().futureValue should ===(Done)
       }
+
+      "some elements not yet delivered to stage" in {
+        val (queue, probe) =
+          Source.queue[Unit](10, OverflowStrategy.fail).toMat(TestSink.probe)(Keep.both).run()
+        intercept[IllegalStateException] {
+          Await.result(
+            (1 to 15).map(_ ⇒ queue.offer(())).last, 3.seconds)
```

Comment:
is this guaranteed to throw?
I think you have to use a TestSink that is not requesting elements to be sure?